### PR TITLE
[hotfix] cancel with savePoint bug fixed.

### PR DESCRIPTION
**Pull request by wolfboys**
[hotfix] cancel with savePoint bug fixed.

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### CODE REVIEW

**Comment by wolfboys** [comment]
@Al-assad PTAL.

File: streamx-plugin/streamx-flink-submit/streamx-flink-submit-core/src/main/scala/com/streamxhub/streamx/flink/submit/trait/FlinkSubmitTrait.scala [hunk 1]
```diff
@@ -388,18 +386,21 @@ trait FlinkSubmitTrait extends Logger {
...
-          .defaultValue(s"${workspace.APP_SAVEPOINTS}")
-      )
-    } else stopRequest.customSavePointPath
+  def getSavePointDir(stopRequest: StopRequest): String = Try(Option(stopRequest.customSavePointPath).get).getOrElse {
+    getOptionFromDefaultFlinkConfig[String](
+      stopRequest.flinkVersion.flinkHome,
+      ConfigOptions.key(CheckpointingOptions.SAVEPOINT_DIRECTORY.key())
+        .stringType()
+        .defaultValue {
+          if (stopRequest.executionMode == ExecutionMode.YARN_APPLICATION) {
```
Comment (SoberChina) [inline streamx-plugin/streamx-flink-submit/streamx-flink-submit-core/src/main/scala/com/streamxhub/streamx/flink/submit/trait/FlinkSubmitTrait.scala]: This logic is wrong，You need to set the parameter `stoprequest#withsavepoint` to check

Comment (wolfboys) [reply streamx-plugin/streamx-flink-submit/streamx-flink-submit-core/src/main/scala/com/streamxhub/streamx/flink/submit/trait/FlinkSubmitTrait.scala]: > This logic is wrong，You need to set the parameter `stoprequest#withsavepoint` to check

thx, fixed.

**Comment by SoberChina** [comment]
Looks good to me.
It looks all right.

**Pull request merged**